compiler-core/src/erlang/pattern.rs

```diff
@@ -31,7 +31,18 @@ fn print<'a>(
 
         Pattern::Discard { .. } => "_".to_doc(),
 
-        Pattern::VarUsage { name, .. } => env.local_var_name(name),
+        Pattern::VarUsage { name, constructor, .. } => {
+            let v = &constructor.as_ref().unwrap().variant;
+            match v {
+                ValueConstructorVariant::ModuleConstant{literal, ..} => {
```

Comment:
Rather than handling just ints we'll need to inline any constant of any type, so we'll need to call the `const_inline` function here.